### PR TITLE
[873] Migrate to Poetry

### LINKED ISSUE #873: Migrate to poetry
Few action items:

- [x] migrate https://github.com/yaptide/yaptide/blob/master/setup.cfg to respective section of pyproject.toml
- [x] move all dependencies (`requirements-*.txt` files) to dependency groups in poetry
- [x] adjust dockerfiles (need installation of poetry as well)
- [x] adjust github actions scripts (https://github.com/yaptide/yaptide/blob/931a588a566613d9e7cf646fa2e20b3928fd43a6/.github/workflows/test-run.yml and others)
- [x] adjust documentation
- [x] adjust gitpod stuff https://github.com/yaptide/yaptide/blob/master/.gitpod.yml

### PR DESCRIPTION
**Pull request by hendzeld**
Resolves #873.

### CODE REVIEW

**Comment by grzanka** [comment]
@hendzeld add poetry to dependabot as well

**Comment by hendzeld** [comment]
> @hendzeld add poetry to dependabot as well

If I read this correctly, it would mean the current configuration is ok.

(https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#package-ecosystem)
obraz


However, I copied the dependencies as they were specified in requirements-*.txt files.
So they probably should be relaxed to `*` or `major releases` (^Y.x.x). Then specific versions should be specified and updated in poetry.lock.

I also need to configure/remove the cache from Docker images.
I will also take a look at caching in CI pipelines.
(https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#about-caching-workflow-dependencies)

**Comment by grzanka** [comment]
> > @hendzeld add poetry to dependabot as well
> 
> If I read this correctly, it would mean the current configuration is ok.
> 
> (https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#package-ecosystem) obraz

That good !

> However, I copied the dependencies as they were specified in requirements-*.txt files. So they probably should be relaxed to `*` or `major releases` (^Y.x.x). Then specific versions should be specified and updated in poetry.lock.

I'd freeze some of the packages as they are now, i.e. we haven't migrated the code to flask==3.0. We can fix the flask at 2.x. I am curious if dependabot would then propose a bump to 3.x or will it only propose bumps at stuff defined in the lock file.

> I also need to configure/remove the cache from Docker images. I will also take a look at caching in CI pipelines. (https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#about-caching-workflow-dependencies)

I think that can be left for subsequent PR.

File: .gitpod.yml [hunk 1]
```diff
@@ -7,9 +7,11 @@ tasks:
       export FLASK_APP=yaptide.application
       export FLASK_DEBUG=1
     init: |
-      pip install -r requirements.txt
+      export POETRY_VIRTUALENVS_CREATE=false
+      export FLASK_SQLALCHEMY_DATABASE_URI=sqlite://
```
Comment (grzanka) [inline .gitpod.yml]: this seems like in-memory database, I think that file-based may be more universal for gitpod. Easier to inspect.

File: Dockerfile-flask [hunk 2]
```diff
@@ -7,9 +8,11 @@ EXPOSE 6000
 ENV FLASK_APP=yaptide.application
 WORKDIR /usr/local/app
 
-# install dependencies
-COPY ./requirements.txt ./
-RUN pip install --no-cache-dir -r requirements.txt
+# Copy poetry configuration files
+COPY poetry.lock poetry.toml pyproject.toml ./
+# Insatll poatry and project dependencies, we also disables virtualenv creation
```
Comment (grzanka) [inline Dockerfile-flask]:
```suggestion
# Install poetry and project dependencies, we also disabled virtualenv creation
```

File: Dockerfile-worker [hunk 3]
```diff
@@ -22,9 +23,11 @@ ENV TOPAS_G4_DATA_DIR="/simulators/geant4_files_path"
 RUN chgrp -R 0 /usr/local/app && \
     chmod -R g=u /usr/local/app
 
-# install dependencies
-COPY requirements.txt .
-RUN pip install --no-cache-dir -r requirements.txt
+# Copy poetry configuration files
+COPY poetry.lock poetry.toml pyproject.toml ./
+# Insatll poatry and project dependencies, we also disables virtualenv creation
```
Comment (grzanka) [inline Dockerfile-worker]: fix grammar

Comment (hendzeld) [reply Dockerfile-worker]: Will be changed to one suggested above

```suggestion
# Install poetry and project dependencies, we also disabled virtualenv creation
```

**Comment by grzanka** [comment]
@hendzeld let me know once I could test it in yap-dev machine

**Comment by hendzeld** [comment]
> > However, I copied the dependencies as they were specified in requirements-*.txt files. So they probably should be relaxed to `*` or `major releases` (^Y.x.x). Then specific versions should be specified and updated in poetry.lock.
> 
> I'd freeze some of the packages as they are now, i.e. we haven't migrated the code to flask==3.0. We can fix the flask at 2.x. I am curious if dependabot would then propose a bump to 3.x or will it only propose bumps at stuff defined in the lock file.

@grzanka 
I created a new repository with a copied configuration. (https://github.com/hendzeld/poetry-dependabot)

Dependabot can update both lock and pyproject files.
I checked documentation and for test I changed its configuration. So it ignores major updates.

Repository: (https://github.com/hendzeld/poetry-dependabot)

File: Dockerfile-flask [hunk 4]
```diff
@@ -1,4 +1,5 @@
 FROM python:3.10-slim
+ARG POETRY_VERSION=1.2.0
```
Comment (grzanka) [inline Dockerfile-flask]: Why 1.2.0 ?
https://pypi.org/project/poetry/#history

**Comment by grzanka** [comment]
Docker fails:

```
ubuntu@yap-dev:~/yaptide$ docker compose up --detach --build
[+] Building 37.9s (8/10)                                                                                                                                               docker:default
 => [yaptide_flask internal] load build definition from Dockerfile-flask                                                                                                          0.0s
 => => transferring dockerfile: 1.22kB                                                                                                                                            0.0s
 => [yaptide_flask internal] load metadata for docker.io/library/python:3.10-slim                                                                                                 0.0s
 => [yaptide_flask internal] load .dockerignore                                                                                                                                   0.0s
 => => transferring context: 2B                                                                                                                                                   0.0s
 => [yaptide_flask 1/6] FROM docker.io/library/python:3.10-slim                                                                                                                   0.0s
 => [yaptide_flask internal] load build context                                                                                                                                   0.1s
 => => transferring context: 767.77kB                                                                                                                                             0.0s
 => CACHED [yaptide_flask 2/6] WORKDIR /usr/local/app                                                                                                                             0.0s
 => [yaptide_flask 3/6] COPY poetry.lock poetry.toml pyproject.toml ./                                                                                                            0.2s
 => ERROR [yaptide_flask 4/6] RUN pip install --no-cache-dir poetry==1.2.0 && poetry install --only main                                                                         37.4s
------                                                                                                                                                                                 
 > [yaptide_flask 4/6] RUN pip install --no-cache-dir poetry==1.2.0 && poetry install --only main:                                                                                     
4.771 Collecting poetry==1.2.0                                                                                                                                                         
4.982   Downloading poetry-1.2.0-py3-none-any.whl (210 kB)                                                                                                                             
5.069      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 210.9/210.9 kB 2.7 MB/s eta 0:00:00                                                                                                 
5.247 Collecting cleo<2.0.0,>=1.0.0a5                                                                                                                                                  
5.284   Downloading cleo-1.0.0a5-py3-none-any.whl (78 kB)
5.291      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 78.7/78.7 kB 91.3 MB/s eta 0:00:00
5.384 Collecting pkginfo<2.0,>=1.5
5.424   Downloading pkginfo-1.10.0-py3-none-any.whl (30 kB)
5.512 Collecting cachecontrol[filecache]<0.13.0,>=0.12.9
5.546   Downloading CacheControl-0.12.14-py2.py3-none-any.whl (21 kB)
5.663 Collecting packaging>=20.4
5.696   Downloading packaging-23.2-py3-none-any.whl (53 kB)
5.703      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 53.0/53.0 kB 85.5 MB/s eta 0:00:00
5.807 Collecting shellingham<2.0,>=1.5
5.840   Downloading shellingham-1.5.4-py2.py3-none-any.whl (9.8 kB)
5.905 Collecting pexpect<5.0.0,>=4.7.0
5.938   Downloading pexpect-4.9.0-py2.py3-none-any.whl (63 kB)
5.944      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 63.8/63.8 kB 88.0 MB/s eta 0:00:00
6.324 Collecting virtualenv
6.358   Downloading virtualenv-20.25.1-py3-none-any.whl (3.8 MB)
6.478      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.8/3.8 MB 33.0 MB/s eta 0:00:00
6.562 Collecting crashtest<0.4.0,>=0.3.0
6.596   Downloading crashtest-0.3.1-py3-none-any.whl (7.0 kB)
6.708 Collecting platformdirs<3.0.0,>=2.5.2
6.741   Downloading platformdirs-2.6.2-py3-none-any.whl (14 kB)
6.913 Collecting jsonschema<5.0.0,>=4.10.0
6.947   Downloading jsonschema-4.21.1-py3-none-any.whl (85 kB)
6.953      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 85.5/85.5 kB 91.4 MB/s eta 0:00:00
7.018 Collecting html5lib<2.0,>=1.0
7.051   Downloading html5lib-1.1-py2.py3-none-any.whl (112 kB)
7.059      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 112.2/112.2 kB 91.9 MB/s eta 0:00:00
7.217 Collecting poetry-core==1.1.0
7.253   Downloading poetry_core-1.1.0-py3-none-any.whl (526 kB)
7.265      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 526.2/526.2 kB 91.3 MB/s eta 0:00:00
7.506 Collecting keyring>=21.2.0
7.538   Downloading keyring-24.3.1-py3-none-any.whl (38 kB)
7.699 Collecting tomlkit!=0.11.2,!=0.11.3,<1.0.0,>=0.11.1
7.732   Downloading tomlkit-0.12.4-py3-none-any.whl (37 kB)
7.897 Collecting cachy<0.4.0,>=0.3.0
7.932   Downloading cachy-0.3.0-py2.py3-none-any.whl (20 kB)
8.113 Collecting requests<3.0,>=2.18
8.146   Downloading requests-2.31.0-py3-none-any.whl (62 kB)
8.152      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.6/62.6 kB 86.8 MB/s eta 0:00:00
8.234 Collecting poetry-plugin-export<2.0.0,>=1.0.6
8.267   Downloading poetry_plugin_export-1.6.0-py3-none-any.whl (10 kB)
9.031 Collecting dulwich<0.21.0,>=0.20.44
9.068   Downloading dulwich-0.20.50-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (499 kB)
9.079      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 500.0/500.0 kB 99.5 MB/s eta 0:00:00
9.280 Collecting urllib3<2.0.0,>=1.26.0
9.323   Downloading urllib3-1.26.18-py2.py3-none-any.whl (143 kB)
9.331      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 143.8/143.8 kB 91.6 MB/s eta 0:00:00
9.408 Collecting requests-toolbelt<0.10.0,>=0.9.1
9.442   Downloading requests_toolbelt-0.9.1-py2.py3-none-any.whl (54 kB)
9.448      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 54.3/54.3 kB 84.7 MB/s eta 0:00:00
9.914 Collecting msgpack>=0.5.2
9.949   Downloading msgpack-1.0.8-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (385 kB)
9.959      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 385.1/385.1 kB 98.6 MB/s eta 0:00:00
10.02 Collecting lockfile>=0.9
10.05   Downloading lockfile-0.12.2-py2.py3-none-any.whl (13 kB)
10.13 Collecting pylev<2.0.0,>=1.3.0
10.17   Downloading pylev-1.4.0-py2.py3-none-any.whl (6.1 kB)
10.33 Collecting six>=1.9
10.37   Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
10.42 Collecting webencodings
10.45   Downloading webencodings-0.5.1-py2.py3-none-any.whl (11 kB)
10.69 Collecting referencing>=0.28.4
10.72   Downloading referencing-0.33.0-py3-none-any.whl (26 kB)
10.80 Collecting jsonschema-specifications>=2023.03.6
10.84   Downloading jsonschema_specifications-2023.12.1-py3-none-any.whl (18 kB)
10.93 Collecting attrs>=22.2.0
10.96   Downloading attrs-23.2.0-py3-none-any.whl (60 kB)
10.97      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 60.8/60.8 kB 90.4 MB/s eta 0:00:00
12.43 Collecting rpds-py>=0.7.1
12.47   Downloading rpds_py-0.18.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.1 MB)
12.49      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.1/1.1 MB 90.1 MB/s eta 0:00:00
12.63 Collecting SecretStorage>=3.2
12.67   Downloading SecretStorage-3.3.3-py3-none-any.whl (15 kB)
12.74 Collecting jaraco.classes
12.77   Downloading jaraco.classes-3.3.1-py3-none-any.whl (6.8 kB)
12.84 Collecting jeepney>=0.4.2
12.87   Downloading jeepney-0.8.0-py3-none-any.whl (48 kB)
12.88      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 48.4/48.4 kB 91.8 MB/s eta 0:00:00
13.13 Collecting importlib-metadata>=4.11.4
13.16   Downloading importlib_metadata-7.0.1-py3-none-any.whl (23 kB)
13.24 Collecting ptyprocess>=0.5
13.27   Downloading ptyprocess-0.7.0-py2.py3-none-any.whl (13 kB)
13.37 Collecting poetry-plugin-export<2.0.0,>=1.0.6
13.40   Downloading poetry_plugin_export-1.5.0-py3-none-any.whl (10 kB)
13.48   Downloading poetry_plugin_export-1.4.0-py3-none-any.whl (10 kB)
13.54   Downloading poetry_plugin_export-1.3.1-py3-none-any.whl (10 kB)
13.63   Downloading poetry_plugin_export-1.3.0-py3-none-any.whl (10 kB)
13.69   Downloading poetry_plugin_export-1.2.0-py3-none-any.whl (10 kB)
13.79   Downloading poetry_plugin_export-1.1.2-py3-none-any.whl (10 kB)
13.97 Collecting certifi>=2017.4.17
14.01   Downloading certifi-2024.2.2-py3-none-any.whl (163 kB)
14.02      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 163.8/163.8 kB 95.7 MB/s eta 0:00:00
14.14 Collecting idna<4,>=2.5
14.17   Downloading idna-3.6-py3-none-any.whl (61 kB)
14.18      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.6/61.6 kB 88.8 MB/s eta 0:00:00
14.78 Collecting charset-normalizer<4,>=2
14.81   Downloading charset_normalizer-3.3.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (142 kB)
14.82      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 142.1/142.1 kB 79.0 MB/s eta 0:00:00
15.09 Collecting virtualenv
15.13   Downloading virtualenv-20.25.0-py3-none-any.whl (3.8 MB)
15.20      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.8/3.8 MB 58.2 MB/s eta 0:00:00
15.34   Downloading virtualenv-20.24.7-py3-none-any.whl (3.8 MB)
15.39      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.8/3.8 MB 89.8 MB/s eta 0:00:00
15.53   Downloading virtualenv-20.24.6-py3-none-any.whl (3.8 MB)
15.58      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.8/3.8 MB 95.1 MB/s eta 0:00:00
15.74   Downloading virtualenv-20.24.5-py3-none-any.whl (3.7 MB)
15.78      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.7/3.7 MB 91.7 MB/s eta 0:00:00
15.93   Downloading virtualenv-20.24.4-py3-none-any.whl (3.7 MB)
15.97      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.7/3.7 MB 94.1 MB/s eta 0:00:00
16.12   Downloading virtualenv-20.24.3-py3-none-any.whl (3.0 MB)
16.15      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.0/3.0 MB 95.1 MB/s eta 0:00:00
16.29   Downloading virtualenv-20.24.2-py3-none-any.whl (3.0 MB)
16.33      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.0/3.0 MB 96.9 MB/s eta 0:00:00
16.47   Downloading virtualenv-20.24.1-py3-none-any.whl (3.0 MB)
16.51      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.0/3.0 MB 83.7 MB/s eta 0:00:00
16.67   Downloading virtualenv-20.24.0-py3-none-any.whl (3.0 MB)
16.70      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.0/3.0 MB 95.4 MB/s eta 0:00:00
16.84   Downloading virtualenv-20.23.1-py3-none-any.whl (3.3 MB)
16.88      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.3/3.3 MB 91.4 MB/s eta 0:00:00
17.02   Downloading virtualenv-20.23.0-py3-none-any.whl (3.3 MB)
17.06      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.3/3.3 MB 93.7 MB/s eta 0:00:00
17.22   Downloading virtualenv-20.22.0-py3-none-any.whl (3.2 MB)
17.26      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.2/3.2 MB 83.5 MB/s eta 0:00:00
17.41   Downloading virtualenv-20.21.1-py3-none-any.whl (8.7 MB)
17.51      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 8.7/8.7 MB 93.3 MB/s eta 0:00:00
17.71 Collecting distlib<1,>=0.3.6
17.74   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
17.75      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 kB 99.8 MB/s eta 0:00:00
17.88 Collecting filelock<4,>=3.4.1
17.91   Downloading filelock-3.13.1-py3-none-any.whl (11 kB)
18.29 Collecting zipp>=0.5
18.33   Downloading zipp-3.17.0-py3-none-any.whl (7.4 kB)
19.80 Collecting cryptography>=2.0
19.84   Downloading cryptography-42.0.5-cp39-abi3-manylinux_2_28_x86_64.whl (4.6 MB)
20.03      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.6/4.6 MB 90.4 MB/s eta 0:00:00
20.26 Collecting more-itertools
20.30   Downloading more_itertools-10.2.0-py3-none-any.whl (57 kB)
20.30      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 57.0/57.0 kB 87.5 MB/s eta 0:00:00
21.14 Collecting cffi>=1.12
21.17   Downloading cffi-1.16.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (443 kB)
21.18      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 443.9/443.9 kB 96.3 MB/s eta 0:00:00
21.36 Collecting pycparser
21.39   Downloading pycparser-2.21-py2.py3-none-any.whl (118 kB)
21.40      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 118.7/118.7 kB 97.1 MB/s eta 0:00:00
22.33 Installing collected packages: webencodings, pylev, ptyprocess, lockfile, distlib, zipp, urllib3, tomlkit, six, shellingham, rpds-py, pycparser, poetry-core, platformdirs, pkginfo, pexpect, packaging, msgpack, more-itertools, jeepney, idna, filelock, crashtest, charset-normalizer, certifi, cachy, attrs, virtualenv, requests, referencing, jaraco.classes, importlib-metadata, html5lib, dulwich, cleo, cffi, requests-toolbelt, jsonschema-specifications, cryptography, cachecontrol, SecretStorage, jsonschema, keyring, poetry-plugin-export, poetry
28.63 Successfully installed SecretStorage-3.3.3 attrs-23.2.0 cachecontrol-0.12.14 cachy-0.3.0 certifi-2024.2.2 cffi-1.16.0 charset-normalizer-3.3.2 cleo-1.0.0a5 crashtest-0.3.1 cryptography-42.0.5 distlib-0.3.8 dulwich-0.20.50 filelock-3.13.1 html5lib-1.1 idna-3.6 importlib-metadata-7.0.1 jaraco.classes-3.3.1 jeepney-0.8.0 jsonschema-4.21.1 jsonschema-specifications-2023.12.1 keyring-24.3.1 lockfile-0.12.2 more-itertools-10.2.0 msgpack-1.0.8 packaging-23.2 pexpect-4.9.0 pkginfo-1.10.0 platformdirs-2.6.2 poetry-1.2.0 poetry-core-1.1.0 poetry-plugin-export-1.1.2 ptyprocess-0.7.0 pycparser-2.21 pylev-1.4.0 referencing-0.33.0 requests-2.31.0 requests-toolbelt-0.9.1 rpds-py-0.18.0 shellingham-1.5.4 six-1.16.0 tomlkit-0.12.4 urllib3-1.26.18 virtualenv-20.21.1 webencodings-0.5.1 zipp-3.17.0
28.63 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
29.04 
29.04 [notice] A new release of pip is available: 23.0.1 -> 24.0
29.04 [notice] To update, run: pip install --upgrade pip
31.54 Skipping virtualenv creation, as specified in config file.
36.79 
36.79 The lock file is not compatible with the current version of Poetry.
36.79 Upgrade Poetry to be able to read the lock file or, alternatively, regenerate the lock file with the `poetry lock` command.
------
failed to solve: process "/bin/sh -c pip install --no-cache-dir poetry==$POETRY_VERSION && poetry install --only main" did not complete successfully: exit code: 1
```

**Review by grzanka — CHANGES_REQUESTED** [review]
https://github.com/yaptide/yaptide/pull/874#issuecomment-1976104488

File: Dockerfile-flask (hunk 4 above)
Comment (hendzeld) [reply Dockerfile-flask]: Right, I will update it to the latest version.
Probably copy/paste mistake.

**Comment by hendzeld** [comment]
@grzanka Should work now.

Works for me :)

**Review by grzanka — APPROVED** [review]

**Pull request merged**